packages/server/lib/controllers/onboarding.controller.ts

```diff
@@ -327,15 +332,19 @@ class OnboardingController {
                 return;
             }
 
+            const progress = req.body.progress;
+
             const { user, account } = response;
             const status = await getOnboardingProgress(user.id);
             if (!status) {
                 res.status(404).send({ message: 'no_onboarding' });
                 return;
             }
 
-            await updateOnboardingProgress(status.id, req.body.progress);
-            void analytics.track(AnalyticsTypes[`DEMO_${req.body.progress as 0}`], account.id, { user_id: user.id });
+            await updateOnboardingProgress(status.id, progress);
+            if (progress === 3 || progress === 6) {
+                void analytics.track(AnalyticsTypes[`DEMO_${progress as 0}`], account.id, { user_id: user.id });
```

Comment:
`progress as 0` ?

Reply:
it's just for typescript sake but actually I can remove this now that I narrow the type number